ospf6d/ospf6_lsa.c

```diff
@@ -1165,6 +1191,23 @@ int config_write_ospf6_debug_lsa(struct vty *vty)
 {
 	unsigned int i;
 	const struct ospf6_lsa_handler *handler;
+	bool debug_all = true;
+
+	for (i = 0; i < vector_active(ospf6_lsa_handler_vector); i++) {
+		handler = vector_slot(ospf6_lsa_handler_vector, i);
+		if (handler == NULL)
+			continue;
+		if (CHECK_FLAG(handler->lh_debug, OSPF6_LSA_DEBUG_ALL)
```

Comment:
Do we really need to do this loop twice? Could the check for `ALL` be added to the existing loop, with a `continue` instead of falling through to the individual categories?

Reply:
Yeah, I think we do. For example if we apply
`debug ospf6 lsa all`
followed by
`no debug ospf6 lsa inter-prefix `
the output should be
```
do show running-config 
Building configuration...

Current configuration:
!
frr version 8.1-dev-MyOwnFRRVersion
frr defaults traditional
hostname frr
no ip forwarding
no ipv6 forwarding
service integrated-vtysh-config
!
debug ospf6 lsa unknown
debug ospf6 lsa unknown originate
debug ospf6 lsa unknown examine
debug ospf6 lsa unknown flooding
debug ospf6 lsa router
debug ospf6 lsa router originate
debug ospf6 lsa router examine
debug ospf6 lsa router flooding
debug ospf6 lsa network
debug ospf6 lsa network originate
debug ospf6 lsa network examine
debug ospf6 lsa network flooding
debug ospf6 lsa inter-prefix
debug ospf6 lsa inter-prefix originate
debug ospf6 lsa inter-prefix flooding
debug ospf6 lsa inter-router
debug ospf6 lsa inter-router originate
debug ospf6 lsa inter-router examine
debug ospf6 lsa inter-router flooding
debug ospf6 lsa as-external
debug ospf6 lsa as-external originate
debug ospf6 lsa as-external examine
debug ospf6 lsa as-external flooding
debug ospf6 lsa nssa
debug ospf6 lsa nssa originate
debug ospf6 lsa nssa examine
debug ospf6 lsa nssa flooding
debug ospf6 lsa link
debug ospf6 lsa link originate
debug ospf6 lsa link examine
debug ospf6 lsa link flooding
debug ospf6 lsa intra-prefix
debug ospf6 lsa intra-prefix originate
debug ospf6 lsa intra-prefix examine
debug ospf6 lsa intra-prefix flooding
debug ospf6 lsa grace
debug ospf6 lsa grace originate
debug ospf6 lsa grace examine
debug ospf6 lsa grace flooding
```
whereas if we keep the check for ALL inside the loop then the output will be (because of the way the debugs are being printed)
```
debug ospf6 lsa inter-prefix
debug ospf6 lsa inter-prefix originate
debug ospf6 lsa inter-prefix flooding
```

Reply:
interesting - that's not what I would expect - maybe that's naive?
If I configure "debug foo", then I don't expect to get "show runn" output that's a lot different from what I configured; I wouldn't expect "debug foo all" to be so different from "debug foo lsa", for example. I was expecting that if you want to configure lots of specific debugs, you ... configure them explicitly, and your config inputs are largely the same as the output of "show running". Do we have other cases where there is an "all" keyword, and it behaves so differently?

Reply:
Even I can't find an example of "all" that is handled this way. My idea behind this was that each LSA type has 3 types (originate|examine|flooding) so we set them all for the "all" case. And in the case when one or more subtypes are unset then we print the remaining subtypes (baring the ones we unset). Similarly, we can unset all using the "no" statement and set particular subtypes explicitly. That was my understanding of the "all" case. Do you have anything else in mind?

Reply:
This is an interesting question ... my inclination is that the operator isn't going to keep a "mental list" of all the debugs turned on or off ... on the other hand, I could see a situation where an operator is troubleshooting, turns on "all," and then sees they're getting too much information, so they start turning off debugs to cut the amount of information they're getting down, so I can see the utility of having this behavior.

I'm inclined to think this is a useful way of building the debugs.